### PR TITLE
fix(ui): move vite plugin to repo

### LINKED ISSUE #9630: Migrate frontend dev dependencies
Migrate following frontend dev dependencies to internal folder for security and legal compliance:

- vite-plugin-i18next-scanner
- vite-plugin-next-react-router

### PR DESCRIPTION
**Pull request by zoubingwu**
### What problem does this PR solve?

Issue Number: close  #9630 

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: dm/ui/plugins/vite-plugin-next-react-router/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  "dependencies": {
+    "consola": "^2.15.3",
+    "fast-glob": "^3.2.7",
+    "lodash": "^4.17.21"
+  },
+  "peerDependencies": {
+    "react-router-dom": ">=6",
+    "vite": ">=2"
+  }
```
Comment (wuhuizuo) [inline dm/ui/plugins/vite-plugin-next-react-router/package.json:21]: should we use persistent patch versions?

Comment (zoubingwu) [reply dm/ui/plugins/vite-plugin-next-react-router/package.json:21]: it's fine here, peerDependencies means it should work with another package and it is compatible with this version range, like a typical plugin system

**Review by wuhuizuo — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Review by lichunzhu — APPROVED** [review]

**Pull request merged**